lib/reply.js

```diff
@@ -244,7 +244,7 @@ Reply.prototype.header = function (key, value = '') {
     }
 
     if (Array.isArray(value)) {
-      this[kReplyHeaders][key].push(...value)
+      Array.prototype.push.apply(this[kReplyHeaders][key], value)
```

Comment:
I believe that `concat` can be even faster. Do you think will be of value?

Reply:
concat is usually slower. 

https://github.com/fastify/fastify/pull/5016?notification_referrer_id=NT_kwDOAE0yHLI3NTkzODc2MDExOjUwNTkxMDA#discussion_r1314246477

Reply:
It is?
I found it usually faster (even with 4 to 5 times its difference), e.g.

```shell
Concat x 27,450,124 ops/sec ±1.12% (96 runs sampled)
Push x 5,495,802 ops/sec ±0.16% (100 runs sampled)
Push#Apply x 5,478,184 ops/sec ±0.20% (97 runs sampled)
Push#PrototypeApply x 5,481,872 ops/sec ±0.17% (102 runs sampled)
Fastest is Concat
```

Code:
```js
const Benchmark = require('benchmark')

const suite = new Benchmark.Suite()

const totalLength = 100
const values = Array.from({ length: totalLength }).map((_, index) => index + 1)

suite
  .add('Concat', function () {
    return [].concat(values);
  })
  .add('Push', function () {
    return [].push(...values);
  })
  .add('Push#Apply', function () {
    let arr = [];
    return values.push.apply(arr, values);
  })
  .add('Push#PrototypeApply', function () {
    return Array.prototype.push.apply([], values);
  })
  .on('cycle', function (event) {
    console.log(String(event.target))
  })
  .on('complete', function () {
    console.log('Fastest is ' + this.filter('fastest').map('name'))
  })
  .run()

```

Unless I did something different 🤔

Reply:
Strange. But concat must(!) be slower, as you have an additional array in memory.

Reply:
On the other hand, push is basically adding each element separately to the array. You can check this by having a huuuge array being added as .prototype.apply([], values) and node will throw an error saying that too many parameters were passed to push. 

So in biased conclusion:
Short arrays to concat should be using .push, humongous arrays should use .concat

Reply:
```shell
Concat x 26,459,581 ops/sec ±1.08% (92 runs sampled)
Concat#Apply x 28,371,954 ops/sec ±1.13% (96 runs sampled)
Concat#PrototypeApply x 26,855,262 ops/sec ±1.48% (94 runs sampled)
Push x 6,274,559 ops/sec ±0.21% (98 runs sampled)
Push#Apply x 6,329,256 ops/sec ±0.15% (94 runs sampled)
Push#PrototypeApply x 6,287,601 ops/sec ±0.27% (99 runs sampled)
Fastest is Concat#Apply
```

Reply:
Let me be loosey-goosey: both code versions are faster than spread operator :D

Reply:
> Strange. But concat must(!) be slower, as you have an additional array in memory.

> On the other hand, push is basically adding each element separately to the array. You can check this by having a huuuge array being added as .prototype.apply([], values) and node will throw an error saying that too many parameters were passed to push.

Just out of curiosity, but I don't really get this, we have 2 arrays in the beginning anyway. What's the difference between adding the elements of one to another 1 by 1 and just concatenating them

If concat is faster in the benches, why not go with that?